src/graphnet/deployment/icecube/inference_module.py

```diff
@@ -0,0 +1,207 @@
+        self._features = features
+
+        # Set GCD file for pulsemap extractor
+        for i3_extractor in self._i3_extractors:
+            i3_extractor.set_gcd(i3_file="", gcd_file=self._gcd_file)
+
+    def __call__(self, frame: I3Frame) -> bool:
+        """Write predictions from model to frame."""
+        # inference
+        data = self._create_data_representation(frame=frame)
+        predictions = self._apply_model(data=data)
+
+        # Check dimensions of predictions and prediction columns
+        dim = self._check_dimensions(predictions=predictions)
+
+        # Build Dictionary from predictions
+        data = self._create_dictionary(dim=dim, predictions=predictions)
+
+        # Submit Dictionary to frame
+        frame = self._add_to_frame(frame=frame, data=data)
+        return True
+
+    def _check_dimensions(self, predictions: np.ndarray) -> int:
+        if len(predictions.shape) > 1:
+            dim = predictions.shape[1]
+        else:
+            dim = len(predictions)
+        try:
+            assert dim == len(self.prediction_columns)
+        except AssertionError as e:
+            self.error(
+                f"predictions have shape {dim} but"
+                f"prediction columns have [{self.prediction_columns}]"
+            )
+            raise e
+
+        assert predictions.shape[0] == 1
+        return dim
+
+    def _create_dictionary(
+        self, dim: int, predictions: np.ndarray
+    ) -> Dict[str, Any]:
+        """Transform predictions into a dictionary."""
+        data = {}
+        for i in range(dim):
+            try:
+                assert len(predictions[:, i]) == 1
+                data[
+                    self.model_name + "_" + self.prediction_columns[i]
+                ] = I3Double(float(predictions[:, i][0]))
+            except IndexError:
+                data[
+                    self.model_name + "_" + self.prediction_columns[i]
+                ] = I3Double(predictions[0])
+        return data
+
+    def _apply_model(self, data: Data) -> np.ndarray:
+        """Apply model to `Data` and case-handling."""
+        if data is not None:
+            predictions = self._inference(data)
+            if isinstance(predictions, list):
+                predictions = predictions[0]
+                self.warning(
+                    f"{self.__class__.__name__} is assumes one Task "
+                    f"but got {len(predictions)}. Only the first will"
+                    " be used."
+                )
+        else:
+            self.warning(
+                "At least one event has no pulses "
+                " - padding {self.prediction_columns} with NaN."
+            )
+            predictions = np.repeat(
+                [np.nan], len(self.prediction_columns)
+            ).reshape(-1, len(self.prediction_columns))
+        return predictions
+
+    def _create_data_representation(
+        self, frame: I3Frame
+    ) -> Data:  # py-l-i-n-t-:- -d-i-s-able=invalid-name
```

Comment:
Is this some auto-correction gone wrong or is the tag supposed to be like that?

Reply:
I believe so. Removed.